### PR TITLE
Fix addImageToTemplate with fullsize

### LINKED ISSUE #1773: addImageToTemplate: undefined method "getName" of class "stdClass"
**Affected version(s)**

Contao 4.9

**Description**

If we use the youtube content element with a splash-image, the function `addImageToTemplate` of the Class `Controller` will be called.
The first parameter of this function is the template object. In the content element this is a `stdClass` object. On line 1701 in the Controller-class the function `getName()` of the tempalte-object is called. But the stdClass-object does not provide such a function.

```php
$strLightboxId = substr(md5($objTemplate->getName() . '_' . $arrItem['id']), 0, 6);
```
`objTemplate` is of type `stdClass`.

https://github.com/contao/contao/blob/master/core-bundle/src/Resources/contao/library/Contao/Controller.php#L1701

What value is expected here?

### PR DESCRIPTION
**Pull request by fritzmg**
| Q                | A
| -----------------| ---
| Fixed issues     | #1773 
| Docs PR or issue | N/A

Currently Contao passes an `stdClass` object to `addImageToTemplate` in three places: `ContentGallery`, `ContentVimeo` and `ContentYouTube`. Within `ContentGallery` it is never a problem, because the `fullsize` attribute will never be in `$arrItem`. However, in the other cases, the content element might contain `fullsize = '1'`, if the content element previously was of type `text` or `image` for example and then later on switched to `youtube` or `vimeo`.

I also added a check for `$arrItem['id']` since that might be empty too.

### CODE REVIEW

File: core-bundle/src/Resources/contao/library/Contao/Controller.php [hunk 1]
```diff
@@ -1695,7 +1695,7 @@ public static function addImageToTemplate($objTemplate, $arrItem, $intMaxWidth=n
 		$objTemplate->picture = $picture;
 
 		// Provide an ID for single lightbox images in HTML5 (see #3742)
-		if ($strLightboxId === null && $arrItem['fullsize'])
+		if ($strLightboxId === null && $arrItem['fullsize'] && $objTemplate instanceof Template && !empty($arrItem['id']))
```
Comment (ausi) [inline core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: Shouldn’t we still generate a lightbox id if it is not a template or doesn’t have an id?

Comment (fritzmg) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: imho it's fine not to. Any custom code that uses `Controller::addImageToTemplate` probably would use the 4th parameter anyway, if they need a lightbox ID.

Comment (m-vo) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: Couldn't we use something that's independent of the template to generate the id?

I'm going to need this as well for #1753. :slightly_smiling_face:

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: I think @ausi has a point here. If we do not generate a lightbox ID, we will end up with an empty `data-lightbox=""` attribute (e.g. see line 1755 or 1792).

Comment (fritzmg) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: Yeah, that is the expected behavior. You only need a light box ID, if you want to group the lightboxes.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: True.

**Review by ausi — APPROVED** [review]

File: core-bundle/src/Resources/contao/library/Contao/Controller.php (hunk 1 above)
Comment (m-vo) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: But all lightboxes with an empty id are then also grouped - would this be OK? (compare https://github.com/contao/core/issues/3742 / comment in line 1697).

Comment (fritzmg) [reply core-bundle/src/Resources/contao/library/Contao/Controller.php:1698]: Hm... true. But the behavior right now isn't different than before.

**Comment by leofeyer** [comment]
Thank you @fritzmg.

**Pull request merged**